plugins/hls-import-actions-plugin/src/Ide/Plugin/ImportActions.hs

```diff
@@ -188,47 +201,58 @@ codeActionProvider _ ideState _pId (CodeActionParams _ _ TextDocumentIdentifier
             , _disabled = Nothing
             , _data_ = data_}
 
-codeActionResolveProvider :: Recorder (WithPriority Log) -> ResolveFunction IdeState EIResolveData 'Method_CodeActionResolve
+codeActionResolveProvider :: Recorder (WithPriority Log) -> ResolveFunction IdeState IAResolveData 'Method_CodeActionResolve
 codeActionResolveProvider _ ideState _ ca _ rd =
   pluginResponse $ do
     wedit <- resolveWTextEdit ideState rd
     pure $ ca & L.edit ?~ wedit
 --------------------------------------------------------------------------------
 
-resolveWTextEdit :: IdeState -> EIResolveData -> ExceptT String (LspT Config IO) WorkspaceEdit
+resolveWTextEdit :: IdeState -> IAResolveData -> ExceptT String (LspT Config IO) WorkspaceEdit
 resolveWTextEdit ideState (ResolveOne uri int) = do
   nfp <- getNormalizedFilePath uri
-  (MinimalImportsResult{forResolve}) <-
-    handleMaybeM "Unable to run Minimal Imports"
+  (ImportActionsResult{forResolve}, pm) <-
+    handleMaybeM "Unable to run Import Actions"
     $ liftIO
-    $ runAction "MinimalImports" ideState $ use MinimalImports nfp
-  tedit <- handleMaybe "Unable to resolve text edit" $ forResolve IM.!? int
-  pure $ mkWorkspaceEdit uri [tedit]
-resolveWTextEdit ideState (ResolveAll uri) = do
+    $ runAction "ImportActions" ideState $ useWithStale ImportActions nfp
+  iEdit <- handleMaybe "Unable to resolve text edit" $ forResolve IM.!? int
+  pure $ mkWorkspaceEdit uri [iEdit] pm
+resolveWTextEdit ideState (ExplicitAll uri) = do
   nfp <- getNormalizedFilePath uri
-  (MinimalImportsResult{forResolve}) <-
-    handleMaybeM "Unable to run Minimal Imports"
+  (ImportActionsResult{forResolve}, pm) <-
+    handleMaybeM "Unable to run Import Actions"
     $ liftIO
-    $ runAction "MinimalImports" ideState $ use MinimalImports nfp
-  let edits = IM.elems forResolve
-  pure $ mkWorkspaceEdit uri edits
-
-mkWorkspaceEdit :: Uri -> [TextEdit] -> WorkspaceEdit
-mkWorkspaceEdit uri edits =
-      WorkspaceEdit {_changes = Just $ Map.fromList [(uri, edits)]
+    $ runAction "ImportActions" ideState $ useWithStale ImportActions nfp
+  let edits = [ ie | ie@ImportEdit{ieResType = ExplicitImport} <- IM.elems forResolve]
+  pure $ mkWorkspaceEdit uri edits pm
+resolveWTextEdit ideState (RefineAll uri) = do
+  nfp <- getNormalizedFilePath uri
+  (ImportActionsResult{forResolve}, pm) <-
+    handleMaybeM "Unable to run Import Actions"
+    $ liftIO
+    $ runAction "ImportActions" ideState $ useWithStale ImportActions nfp
```

Comment:
I forget, are we generally using `useWithStale` in the resolve handlers, or is it better to use `use`?